alf/utils/math_ops.py

```diff
@@ -198,3 +198,58 @@ def shuffle(values):
     batch_size = alf.nest.get_nest_batch_size(values)
     indices = torch.randperm(batch_size)
     return nest_map(lambda value: value[indices], values)
+
+
+class Softsign_(torch.autograd.Function):
+    r"""softsign(input) -> Tensor
+
+    Applies element-wise inplace, the function :math:`\text{SoftSign}(x) = \frac{x}{1 + |x|}`
+    
+    The `current pytorch implementation of softsign
+    <https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#softsign>`_
+    is inefficient for backward and does not have an inplace version. Hence we
```

Comment:
The reason why the current softsign is inefficient is because there is no explicit gradient defined?

Reply:
Right. It relies on AD to do it. And because it involve several steps of computation. All the intermediate result need to be kept.